### PR TITLE
CM-24996 - Integrate Black code formatter

### PR DESCRIPTION
**Pull request by MarshalX**

### CODE REVIEW

File: .github/workflows/black.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.7
+
+      - name: Setup Poetry
+        uses: snok/install-poetry@v1
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply .github/workflows/black.yml:30]: #cycode_ignore_insecure_pipeline_violation_here

File: poetry.lock [hunk 2]
```diff
@@ -394,13 +455,13 @@ files = [
 
 [[package]]
 name = "pathspec"
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply poetry.lock:457]: #cycode_ignore_non_permissive_license_use

File: cycode/cyclient/cycode_client_base.py [hunk 3]
```diff
@@ -42,48 +42,24 @@ def reset_user_agent() -> None:
...
     ) -> Response:
         url = self.build_full_url(self.api_url, endpoint)
         logger.debug(f'Executing {method.upper()} request to {url}')
 
         try:
             headers = self.get_request_headers(headers, without_auth=without_auth)
-            response = request(
-                method=method, url=url, timeout=self.timeout, headers=headers, **kwargs
-            )
+            response = request(method=method, url=url, timeout=self.timeout, headers=headers, **kwargs)
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply cycode/cyclient/cycode_client_base.py:62]: #cycode_sast_ignore_here false positive; proof:

https://github.com/cycodehq-public/cycode-cli/blob/657c7fc2497dccbdfb08981508d9aa450b5a143b/cycode/cyclient/cycode_client_base.py#L84-L90

File: cycode/cli/helpers/maven/base_restore_maven_dependencies.py [hunk 4]
```diff
@@ -54,8 +54,9 @@ def get_lock_file_name(self) -> str:
...
     def try_restore_dependencies(self, document: Document) -> Optional[Document]:
         manifest_file_path = self.get_manifest_file_path(document)
-        document = Document(build_dep_tree_path(document.path, self.get_lock_file_name()),
-                            execute_command(self.get_command(manifest_file_path), manifest_file_path,
-                                            self.command_timeout),
-                            self.is_git_diff)
+        document = Document(
+            build_dep_tree_path(document.path, self.get_lock_file_name()),
+            execute_command(self.get_command(manifest_file_path), manifest_file_path, self.command_timeout),
+            self.is_git_diff,
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply cycode/cli/helpers/maven/base_restore_maven_dependencies.py:60]: looks false positive. all functions called #cycode_sast_ignore_here

File: cycode/cli/helpers/maven/restore_maven_dependencies.py [hunk 5]
```diff
@@ -31,24 +33,23 @@ def try_restore_dependencies(self, document: Document) -> Optional[Document]:
...
-        backup_restore_content = execute_command(
-            secondary_restore_command,
-            manifest_file_path,
-            self.command_timeout)
-        restore_dependencies_document = Document(build_dep_tree_path(document.path, MAVEN_DEP_TREE_FILE_NAME),
-                                                 backup_restore_content,
-                                                 self.is_git_diff)
+        backup_restore_content = execute_command(secondary_restore_command, manifest_file_path, self.command_timeout)
+        restore_dependencies_document = Document(
+            build_dep_tree_path(document.path, MAVEN_DEP_TREE_FILE_NAME), backup_restore_content, self.is_git_diff
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply cycode/cli/helpers/maven/restore_maven_dependencies.py:51]: looks false positive. all functions called #cycode_sast_ignore_here

[the remark this answers is not in the document]
Comment (MarshalX) [reply cycode/cli/helpers/maven/restore_maven_dependencies.py:51]: looks false positive. all functions called #cycode_sast_ignore_here

[the remark this answers is not in the document]
Comment (MarshalX) [reply cycode/cli/helpers/maven/restore_maven_dependencies.py:51]: looks false positive. all functions called #cycode_sast_ignore_here

**Review by artem-fedorov — APPROVED** [review]

**Pull request merged**